### ISSUE TITLE
specviz crash: model fitting

### ISSUE THREAD

**Issue by gderosa2004** [spacetelescope/specviz#479]
If I load a spectrum (e.g. example_stis.fits), select a region, click on add model and select Const1d specviz crashes with the following message

```
Traceback (most recent call last):
  File "/Users/gderosa/Desktop/specviz/specviz/plugins/model_editor/model_editor.py", line 41, in <lambda>
    action.triggered.connect(lambda x, m=v: self._add_fittable_model(m))
  File "/Users/gderosa/Desktop/specviz/specviz/plugins/model_editor/model_editor.py", line 88, in _add_fittable_model
    idx = self.model_tree_view.model().add_model(model())
AttributeError: 'NoneType' object has no attribute 'add_model'
Abort trap: 6
```

**Comment by brechmos-stsci** [comment]
@robelgeda is working on the fitting part, so this one might have to wait and probably could be associated with https://github.com/spacetelescope/specviz/issues/414

**Comment by nmearl** [comment]
This is actually due to the fact that a model-based spectrum object hasn't yet been created. That is, currently, you'd need to select the data in the list, click the "New Model" button at the top. This creates a new data item in the list that is model-based. You'd then add new sub models to this model-based spectrum.

**Comment by nmearl** [comment]
Certainly the plugin should be disabled until the user has created a new model spectrum, or selects an already created one.

**Comment by nmearl** [comment]
#500 now disallows users from preemptively adding submodels.

<img width="1392" alt="screen shot 2018-10-29 at 12 50 27 pm" src="https://user-images.githubusercontent.com/4141126/47666240-6bbeeb00-db79-11e8-84ab-c0640c2b9839.png">

**Comment by hcferguson** [comment]
Addressed by #500.
